### PR TITLE
feat: docarray vectorstore

### PR DESCRIPTION
**Pull request by jupyterjazz**
on the outside, will look like this:

```python
vector_base = VectorBase('docarray')
```

waiting for the docarray release and will push a PR

### CODE REVIEW

File: gptcache/manager/vector_data/in_memory_exact_search.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            DocarrayVectorData(id=data.id, data=data.data) for data in datas
+        )
+        self._index.index(docs)
+
+    def search(self, data: np.ndarray, top_k: int = -1):
+        query = parse_obj_as(NdArray, data)
+        docs, scores = self._index.find(query, search_field="data", limit=top_k)
+        return list(zip(scores, docs.id))
+
+    def rebuild(self, ids=None):
```
Comment (JoanFM) [inline gptcache/manager/vector_data/in_memory_exact_search.py]: why is this needed?

Comment (jupyterjazz) [reply gptcache/manager/vector_data/in_memory_exact_search.py]: This function is used to rebuild the index with a new data, represented by IDs. It's used when some number of deletions happen and the index needs to be updated to reflect the changes. But deletions doesn't affect InMemory index data structure so we can ignore it.

Comment (jupyterjazz) [reply gptcache/manager/vector_data/in_memory_exact_search.py]: We could also add this to our indices, hnsw will definitely benefit from it

**Review by samsja — CHANGES_REQUESTED** [review]
we need to keep the single quote as it was originally

**Review by samsja — CHANGES_REQUESTED** [review]
we need to keep the single quote as it was originally

**Pull request merged**